odoo/modules/registry.py

```diff
@@ -325,71 +324,134 @@ def field_computed(self):
                                     model_name, ", ".join(field.name for field in fields))
         return computed
 
-    @lazy_property
-    def field_triggers(self):
-        # determine field dependencies
-        dependencies = {}
-        for Model in self.models.values():
-            if Model._abstract:
-                continue
-            for field in Model._fields.values():
-                # dependencies of custom fields may not exist; ignore that case
-                exceptions = (Exception,) if field.base_field.manual else ()
-                with suppress(*exceptions):
-                    dependencies[field] = OrderedSet(field.resolve_depends(self))
-
-        # determine transitive dependencies
-        def transitive_dependencies(field, seen=[]):
-            if field in seen:
+    def get_trigger_tree(self, fields: list, select=bool) -> "TriggerTree":
+        """ Return the trigger tree to traverse when ``fields`` have been modified.
+        The function ``select`` is called on every field to determine which fields
+        should be kept in the tree nodes.  This enables to discard some unnecessary
+        fields from the tree nodes.
+        """
+        trees = [
+            self.get_field_trigger_tree(field)
+            for field in fields
+            if field in self._field_triggers
+        ]
+        return TriggerTree.merge(trees, select)
+
+    def get_dependent_fields(self, field):
+        """ Return an iterator on the fields that depend on ``field``. """
+        return (
+            field
+            for tree in self.get_field_trigger_tree(field).depth_first()
+            for field in tree.root
```

Comment:
-> `field_root` because `field` is also the parameter name (IMO, it is confusing).